core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -122,24 +143,32 @@ private Schema initializeOutputSchema() {
    * @return true
    */
   private void validateInputFields(Schema inputSchema) {
-    if ((inputSchema.getField(conf.getKeyField())) == null) {
+
+    Schema.Field keyField = inputSchema.getField(conf.getKeyField());
+    Schema.Field fieldName = inputSchema.getField(conf.getFieldName());
```

Comment:
I've never really liked the naming for 'fieldName' and 'fieldValue', as they sound like they are just the name and value of a field. Can we change to 'nameField' and 'valueField' (both here and in the plugin config as well)?  I think it is clearer that way, and it also makes it consistent with 'keyField'.